pkg/controller/jobframework/reconciler.go

```diff
@@ -535,10 +535,13 @@ func (r *JobReconciler) ReconcileGenericJob(ctx context.Context, req ctrl.Reques
 			if !job.IsActive() {
 				log.V(6).Info("The job is no longer active, clear the workloads admission")
 				// The requeued condition status set to true only on EvictedByPreemption
+				wlOrig := wl.DeepCopy()
 				setRequeued := evCond.Reason == kueue.WorkloadEvictedByPreemption
 				workload.SetRequeuedCondition(wl, evCond.Reason, evCond.Message, setRequeued)
 				_ = workload.UnsetQuotaReservationWithCondition(wl, "Pending", evCond.Message, r.clock.Now())
```

Comment:
I'm wondering if we can move it inside customPrepare function and check if it is changes.